src/subscription_manager/managercli.py

```diff
@@ -2604,11 +2604,9 @@ def _do_command(self):
         on_date = None
         if self.options.on_date:
             try:
-                on_date = datetime.datetime.strptime(self.options.on_date, '%Y-%m-%d')
-                if on_date.date() < datetime.datetime.now().date():
-                    system_exit(os.EX_USAGE, _("Past dates are not allowed"))
-            except Exception:
-                system_exit(os.EX_DATAERR, _("Date entered is invalid. Date should be in YYYY-MM-DD format (example: ") + strftime("%Y-%m-%d", localtime()) + " )")
+                on_date = entitlement.EntitlementService.parse_date(self.options.on_date)
+            except ValueError as err:
+                system_exit(os.EX_DATAERR, err)
```

Comment:
@jsefler I changed returned error codes here. Subman returns 65 in both cases. Could it be issue? In case it could be issue, then please let me know and I will implement special Exceptions. Thus it will be possible to return different error codes for both cases.

Reply:
If I understand correctly, the only change is that EX_DATAERR (65) will also be the exit code when a user provides an on_date value in the past.  This is okay with me.  I have one test that I will need to update after this pull request is merged.  Thank you for the update.